userspace/libsinsp/gen_filter.h

```diff
@@ -246,21 +246,48 @@ class gen_event_filter_factory
 
 		// A description of the field
 		std::string desc;
+
+		// The data type for the field
+		std::string data_type;
+
+		// A set of free-form tags for the field. Examples include:
+		// FILTER ONLY: for fields that can only be used in filters, not outputs.
+		// IDX_REQUIRED: for fields that can take an optional index
+		// etc
+		std::set<std::string> tags;
 	};
 
-	// A struct describing a group of filtercheck fields ("ka")
-	struct filter_fieldclass_info
+	// Describes a group of filtercheck fields ("ka")
+	class filter_fieldclass_info
 	{
+	public:
 		// The name of the group of fields
 		std::string name;
 
-		// A short description for the fields
+		// A description for the fields
 		std::string desc;
 
-		// Additional information about proper use of the fields
-		std::string class_info;
+		// A short (< 10 words) description of the fields. Can be blank.
+		std::string shortdesc;
 
 		std::list<filter_field_info> fields;
+
+		// Print a terminal-friendly representation of this
+		// field class, including name, description, supported
+		// event sources, and the name and description of each field.
+		std::string as_string(bool verbose, std::set<std::string> event_sources);
+
+		// Calls above with an empty set of event sources.
+		std::string as_string(bool verbose);
```

Comment:
Why just not using a default value for above 
`std::string as_string(bool verbose, const std::set<std::string>& event_sources = std::set<std::string>());`

Reply:
Moreover, i'd embed the markdown representation there, ie:
`std::string as_string(bool verbose, bool markdown = false, const std::set<std::string>& event_sources = std::set<std::string>());`

Reply:
Thanks! +1 from me